plugins/web-worker-offloading/helper.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * Helpers for Web Worker Offloading.
+ *
+ * @since 0.1.0
+ * @package web-worker-offloading
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Gets configuration for Web Worker Offloading.
+ *
+ * @since 0.1.0
+ * @link https://partytown.builder.io/configuration
+ * @link https://github.com/BuilderIO/partytown/blob/b292a14047a0c12ca05ba97df1833935d42fdb66/src/lib/types.ts#L393-L548
+ *
+ * @return array<string, mixed> Configuration for Partytown.
+ */
+function wwo_get_configuration(): array {
+	$config = array(
+		'lib' => wp_parse_url( plugin_dir_url( __FILE__ ), PHP_URL_PATH ) . 'build/',
+	);
+
+	if ( WP_DEBUG && SCRIPT_DEBUG ) {
+		$config['debug'] = true;
+	}
+
+	/**
+	 * Add configuration for Web Worker Offloading.
+	 *
+	 * Many of the configuration options are not documented publicly, so refer to the TypeScript definitions.
+	 * Additionally, not all of the configuration options (e.g. functions) can be serialized as JSON and must instead be
+	 * defined in JavaScript instead. To do so, use the following PHP code instead of filtering `wwo_configuration`:
+	 *
+	 *     add_action(
+	 *         'wp_enqueue_scripts',
+	 *         function () {
+	 *             wp_add_inline_script(
+	 *                 'web-worker-offloading',
+	 *                 <<<JS
+	 *                 window.partytown = {
+	 *                     ...(window.partytown || {}),
+	 *                     resolveUrl: (url, location, type) => {
+	 *                         if (type === 'script') {
+	 *                             const proxyUrl = new URL('https://my-reverse-proxy.example.com/');
+	 *                             proxyUrl.searchParams.append('url', url.href);
+	 *                             return proxyUrl;
+	 *                         }
+	 *                         return url;
+	 *                     },
+	 *                 };
+	 *                 JS,
+	 *                 'before'
+	 *             );
+	 *         }
+	 *     );
+	 *
+	 * There are also many configuration options which are not documented, so refer to the TypeScript definitions.
+	 *
+	 * @since 0.1.0
+	 * @link https://partytown.builder.io/configuration
+	 * @link https://github.com/BuilderIO/partytown/blob/b292a14047a0c12ca05ba97df1833935d42fdb66/src/lib/types.ts#L393-L548
+	 *
+	 * @param array<string, mixed> $config Configuration for Partytown.
+	 */
+	return (array) apply_filters( 'wwo_configuration', $config );
```

Comment:
I think it would be good to add some sanitization here.

For example, since several supported properties/keys are arrays and several plugins/sources may add similar entries to them, we could sanitize these arrays using `array_unique()`. Just thinking about what this PR implements for WooCommerce for example allowlisting `wp`, I'm sure lots of scripts would need that.

Reply:
I would agree with you, but I think sanitizing for this initial version could go down a rabbit hole. 

The number of possible configuration options is large and the value types can be complex: https://github.com/BuilderIO/partytown/blob/main/src/lib/types.ts

We could implement a JSON Schema to mirror the TypeScript types and use this to provide usage warnings. However, I think this validation would be better done inside Partytown itself where the types are being defined.

In the case of `mainWindowAccessors`, there's no need for `array_unique()` because Partytown is using the value like [this](https://github.com/BuilderIO/partytown/blob/b292a14047a0c12ca05ba97df1833935d42fdb66/src/lib/web-worker/worker-window.ts#L370-L371):

```js
              } else if (webWorkerCtx.$config$.mainWindowAccessors?.includes(propName)) {
                return getter(this, [propName]);
```

So I think we should consider what validation and sanitization we do as part of this plugin for a subsequent release.

Reply:
I think it depends also on how _we_ want to use their configuration options. We don't need to holistically capture everything Partytown supports, that would indeed be a rabbit hole. But for instance, it seems that `mainWindowAccessors` will be relevant in how we would use Partytown for quite a few integrations, so we could do some sanitization there where it makes sense. In other words, I think we should do this based on demand.

As long as all of these 4 arguments only support an array of strings, and as long as duplicate values are not a problem, it's probably okay not to sanitize anything. So we can move ahead with this as is, but I think we should look out for when things are more complicated to use, or could be of more than one type, and especially then we should do some sanitization for how _we_ want those arguments to be used.